operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -211,206 +225,248 @@ func (r *RedpandaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	if err := validateClusterParameters(rp); err != nil {
-		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonTerminalError, err.Error())
-
-		logger.Error(err, "validating cluster parameters")
-		r.EventRecorder.Eventf(rp, "Warning", redpandav1alpha2.EventSeverityError, err.Error())
-		return r.syncStatus(ctx, status, cluster)
-	}
+	reconcilers := []clusterReconciliationFn{
+		r.reconcileParameterValidation,
+		// we sync all our non pool resources first so that they're in-place
+		// prior to us scaling up our node pools
+		r.reconcileResources,
+		// next we sync up all of our pools themselves
+		r.reconcilePools,
+		// now we memoize the admin client onto the state
+		r.getAdminClient,
+		// now we ensure that we reconcile all of our decommissioning nodes
+		// TODO: Do we want to rate limit this as well given that it also calls the admin API?
+		// My thought is no since we want to be snappy with decommissioning.
+		r.reconcileDecommission,
+		// now reconcile cluster configuration
+		r.reconcileClusterConfig,
+		// finally reconcile all of our license information
+		r.reconcileLicense,
+	}
+
+	for _, reconciler := range reconcilers {
+		result, err := reconciler(ctx, state)
+		// if we have an error or an explicit requeue from one of our
+		// sub reconcilers, then just early return
+		if err != nil || result.Requeue || result.RequeueAfter > 0 {
+			return r.syncStatus(ctx, state, result, err)
+		}
+	}
+
+	// we're at the end of reconciliation, so sync back our status
+	return r.syncStatus(ctx, state, ctrl.Result{}, nil)
+}
 
-	// we sync all our non pool resources first so that they're in-place
-	// prior to us scaling up our node pools
-	if err := r.reconcileResources(ctx, cluster); err != nil {
-		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonError, err.Error())
+func (r *RedpandaReconciler) fetchInitialState(ctx context.Context, rp *redpandav1alpha2.Redpanda) (*clusterReconciliationState, error) {
+	logger := log.FromContext(ctx)
 
-		logger.Error(err, "error reconciling resources")
-		return r.syncStatusErr(ctx, err, status, cluster)
+	// fetch any related node pools
+	var err error
+	var existingPools []*redpandav1alpha2.NodePool
+	if r.UseNodePools {
+		existingPools, err = fromSourceCluster(ctx, r.Client, "pool", rp, &redpandav1alpha2.NodePoolList{})
+		if err != nil {
+			logger.Error(err, "fetching desired node pools")
+			return nil, err
+		}
 	}
 
-	// next we sync up all of our pools themselves
-	requeue, err := r.reconcilePools(ctx, cluster, pools)
-	if err != nil {
-		status.Status.SetResourcesSynced(statuses.ClusterResourcesSyncedReasonError, err.Error())
+	rp.ManagedFields = nil // nil out our managed fields
```

Comment:
I think this is automatically handled now that we're using the Ctl in most places.

Reply:
I’ll remove it and see if anything breaks 🎉